force-app/classes/test/MethodSpyTest.cls

```diff
@@ -43,26 +44,27 @@ private class MethodSpyTest {
   static void givenSpyConfiguredWithoutArguments_whenCalledWithoutArgument_spyIsCalled() {
     // Arrange
     final MethodSpy sut = new MethodSpy('methodName');
-    sut.whenCalledWith(new List<Object>()).thenReturn('test');
+    sut.whenCalledWith(Params.of()).thenReturn('test');
 
     // Act
     final Object result = sut.call(new List<Object>());
 
     // Assert
     System.assertEquals('test', result);
-    System.assertEquals(true, sut.hasBeenCalledWith(new List<Object>()));
-    System.assertEquals(true, sut.hasBeenLastCalledWith(new List<Object>()));
+    System.assertEquals(true, sut.hasBeenCalledWith(Params.of()));
+    System.assertEquals(true, sut.hasBeenLastCalledWith(Params.of()));
   }
 
   @IsTest
   static void givenSpyConfiguredWithAMixOfObjectAndParamMatcher_whenCalledWithAllMatching_spyIsCalled() {
     // Arrange
     final MethodSpy sut = new MethodSpy('methodName');
-    List<Object> listOfArgs = new List<Object>{
+    Params listOfArgs = Params.of(
```

Comment:
nitpick: finalize this variable

Reply:
suggestion: move declaration next to its usage and rename it `expectedArgs`(in `// Assert` section)

(note: same suggestion for the whole file)